src/main/webapp/app/atlas/manage/agent-chat-modal/agent-chat-modal.component.html

```diff
@@ -20,7 +20,67 @@ <h4 class="modal-title mb-0 text-body" jhiTranslate="artemisApp.agent.chat.title
-                        <div class="message-text">{{ message.content }}</div>
+                        @if (message.content) {
+                            <div class="message-text">{{ message.content }}</div>
+                        }
+                        @if (message.planPending && !message.planApproved) {
+                            <div class="plan-approval-container mt-2">
+                                <button type="button" class="btn btn-primary btn-sm" (click)="onApprovePlan(message)" [disabled]="isAgentTyping()">Approve Plan</button>
+                            </div>
+                        }
+                        @if (message.competencyPreview) {
+                            <div class="competency-preview-container mt-2">
+                                <jhi-competency-card [courseId]="courseId" [competency]="message.competencyPreview" [noProgressRings]="true" />
+                                @if (!message.competencyPreview.viewOnly) {
+                                    <button
+                                        type="button"
+                                        class="btn btn-sm mt-2"
+                                        [class.btn-primary]="!message.competencyCreated"
+                                        [class.btn-success]="message.competencyCreated"
+                                        (click)="onCreateCompetency(message)"
+                                        [disabled]="isAgentTyping() || message.competencyCreated"
+                                    >
+                                        @if (message.competencyCreated) {
+                                            {{ message.competencyPreview.competencyId ? 'Updated' : 'Created' }}
+                                        } @else {
+                                            {{ message.competencyPreview.competencyId ? 'Update' : 'Create' }}
+                                        }
+                                    </button>
+                                }
+                            </div>
+                        }
+                        @if (message.batchCompetencyPreview && message.batchCompetencyPreview.length > 0) {
+                            <div class="batch-competency-preview-container mt-2">
+                                <div class="mb-2">
+                                    <small class="text-muted">Preview: {{ message.batchCompetencyPreview.length }} competencies</small>
+                                </div>
+                                @for (comp of message.batchCompetencyPreview; track comp.title + comp.competencyId) {
+                                    <div class="mb-2">
+                                        <jhi-competency-card [courseId]="courseId" [competency]="comp" [noProgressRings]="true" />
+                                    </div>
+                                }
+                                @if (!message.batchCompetencyPreview[0]?.viewOnly) {
```

Comment:
Can we give the condition a descriptive name here to implicitly explain to the reader what you are checking for? Something like:
```suggestion
                                @let notEartheIsRound = !message.batchCompetencyPreview[0]?.viewOnly
                                @if (notEarthIsRound) {
```